src/commands/validate-release-version.yml

```diff
@@ -9,26 +9,29 @@ description: >
 
 steps:
   - jq/install
+  - install-standard-version
   - run:
       name: Check if version already exists in registry
       command: |
         PKG_NAME=$(cat package.json | jq -r .name)
         PKG_VERSION=$(cat package.json | jq -r .version)
         VERSIONS=$(npm view $PKG_NAME versions --json | jq .)
+        EXISTS_IN_REGISTRY=$(echo $VERSIONS | jq "map(select(. == \"${PKG_VERSION}\")) | length")
 
-        echo "Version we want to release: ${PKG_VERSION}"
-        echo "Existing versions of ${PKG_NAME}:"
-        echo $VERSIONS
+        # if PKG_VERSION does NOT exist in registry, then we can assume that it's the version we want published
+        if (( $EXISTS_IN_REGISTRY == 0 )); then
+            echo "${PKG_NAME}@${PKG_VERSION} (as specifed in the package.json) does not exist in the registry. Assuming that we want to publish it"
 
-        # Searches array of versions for one that matches the version we're attempting to publish in the job
-        # If the version already exists, MATCHING_VERSIONS will be >= 1 otherwise it will be 0
-        MATCHING_VERSIONS=$(echo $VERSIONS | jq "map(select(. == \"${PKG_VERSION}\")) | length")
-        echo "Number of matching versions: ${MATCHING_VERSIONS}"
-        if (( $MATCHING_VERSIONS >= 1 )); then
-            echo "${PKG_NAME}@${PKG_VERSION} already exists in the npm registry..."
-            echo "Exiting release"
-            exit 1
+        # if PKG_VERSION does exist in the registry, then we assume that standard-version will handle the bump
         else
-          echo "${PKG_NAME}@${PKG_VERSION} does not exist in the npm registry..."
-          echo "Proceeding with release"
+            VERSION_TO_BE_PUBLISHED=$(standard-version --dry-run --skip.tag --skip.commit --skip.changelog | grep -o 'to\s.*' | awk '{print $2}')
+            echo "Verifying that version determined by standard-version (${VERSION_TO_BE_PUBLISHED}) does not exist in registry"
+            EXISTS_IN_REGISTRY=$(echo $VERSIONS | jq "map(select(. == \"${VERSION_TO_BE_PUBLISHED}\")) | length")
+            if (( $EXISTS_IN_REGISTRY == 0 )); then
+                echo "${PKG_NAME}@${VERSION_TO_BE_PUBLISHED} does not exist in the registry. Proceeding..."
+                exit 0
+            else
+                echo "${PKG_NAME}@${VERSION_TO_BE_PUBLISHED} already exists in the registry. Exiting..."
```

Comment:
What is the case where this will happen?

Reply:
It could happen if someone
- published 1.0.0
- reverted the version to 0.9.0
- deleted 1.0.0 from the npm registry
- tried publishing 1.0.0 again

Or if someone published their package from the command line and didn't push up the changes to package.json

Both are unlikely but I thought this was a situation where it doesn't hurt to check